test-cypress/integration/explore/class.hierarchy.spec.js

```diff
@@ -1,180 +1,179 @@
-            .each(($element) => {
-                // let display = ;
-                let parsed = $element.prop('style');
-                if (parsed.display != 'none') {
-                    expect($element.text()).contain(':');
-                }
+    function getCurrentSliderValue() {
+        // The count is taken from the rz-pointer's attribute and not from a visible in the UI value
+        // as the rz-slider library doesn't provide a reliable way to get this. It just has multiple
+        // '.rz-bubble' elements and no appropriate selector for the one which holds the visible
+        // value.
+        return cy.get('.rz-pointer[role="slider"]')
+            .then(($element) => $element.prop('aria-valuenow'));
+    }
+
+    function searchForClass(name) {
+        cy.get('.toolbar-holder .icon-search')
+            .click()
+            .then(() => {
+                cy.get('#search_input_value').type(name).type('{enter}').wait(500);
             });
+    }
 
-        // Switch show prefixes to off
-        cy.get('button[ng-click="hidePrefixes = !hidePrefixes"]').click().wait(200);
-
-        // Verify that prefixes are removed from diagram
-        cy.get('#main-group > text.label')
+    function verifyPrefixes(expectation) {
+        cy.get(CLASS_LABEL_SELECTOR)
             .each(($element) => {
-                // let display = ;
-                let parsed = $element.prop('style');
-                if (parsed.display != 'none') {
-                    expect($element.text()).not.contain(':');
+                if ($element.prop('style').display !== 'none') {
+                    expectation($element);
                 }
             });
-    });
+    }
 
-    it('Test focus on diagram', function () {
-        let className = ':CabernetSauvignon';
-        // Select a class ":CabernetSauvignon"
-        cy.get('#main-group > text.label')
+    function findClassByName(className) {
+        cy.get(CLASS_LABEL_SELECTOR)
             .each(($element) => {
                 let data = $element.prop('__data__');
                 if (data.name === className) {
-                    let fontSize = $element.prop('style');
-                    expect(fontSize['font-size']).be.eq('10px');
+                    cy.wrap($element).as('classInHierarchy');
                 }
             });
-        // Verify that the diagram zooms that class
-        cy.get('.p-0 > .icon-search')
-            .click()
-            .then(() => {
-                cy.get('#search_input_value').type(className).type('{enter}').wait(500);
+    }
+
+    function verifyClassIsNotExpanded($element) {
+        let fontSize = $element.prop('style');
+        return expect(fontSize['font-size']).be.eq('10px');
+    }
+
+    function verifyClassIsExpanded($element) {
+        let fontSize = $element.prop('style');
+        return expect(fontSize['font-size']).not.be.eq('10px');
+    }
+
+    function confirmReload() {
+        cy.get('.modal-footer .confirm-btn').click();
```

Comment:
I wonder if ng-animate would case issues in this spec.

Reply:
It didn't see it failing.